firmware/defmt-test/macros/src/lib.rs

```diff
@@ -181,9 +194,13 @@ fn tests_impl(args: TokenStream, input: TokenStream) -> parse::Result<TokenStrea
         } else {
             quote!(#ident())
         };
-        unit_test_calls.push(quote!(
-            #krate::export::check_outcome(#call, #should_error);
-        ));
+        if ignore {
+            unit_test_calls.push(quote!(let _ = #call;));
```

Comment:
This will still call the test function if it has `#[ignore]` on it, leading to the issue reported in https://github.com/knurling-rs/defmt/issues/731